src/geometry.rs

```diff
@@ -166,6 +166,16 @@ impl Geometry {
     pub fn normals(&self) -> &[Vector3<f32>] {
         &self.normals
     }
+
+    pub fn edges(&self) -> Vec<Edge> {
+        self.faces
+        .iter()
+        .flat_map(|face| 
+            match face {
+                Face::Triangle(triangle_face) => triangle_face.to_edges()
+            }
+        ).collect()
```

Comment:
This could just return a borrowed iterator. See `triangle_faces_iter()` and `triangle_faces_len()`.

Reply:
giving up...
```
pub fn edges(&self) -> Vec<Edge> {
        self.triangle_faces_iter()
            .flat_map(|face| face.to_edges().iter())
            .collect()
    }
```
console:
```
error[E0277]: a collection of type `std::vec::Vec<geometry::Edge>` cannot be built from an iterator over elements of type `&geometry::Edge`
   --> src\geometry.rs:173:14
    |
173 |             .collect()
    |              ^^^^^^^ a collection of type `std::vec::Vec<geometry::Edge>` cannot be built from `std::iter::Iterator<Item=&geometry::Edge>`
    |
    = help: the trait `std::iter::FromIterator<&geometry::Edge>` is not implemented for `std::vec::Vec<geometry::Edge>`
```

Reply:
@janper 
```rust
pub fn edges<'a>(&'a self) -> impl Iterator<Item = Edge> + 'a {      
    self.faces.iter().copied().flat_map(|face| match face {
        Face::Triangle(triangle_face) => triangle_face.to_edges(),
    })
}
```

Reply:
```
error[E0277]: `[geometry::Edge; 3]` is not an iterator
   --> src\geometry.rs:177:36
    |
177 |         self.faces.iter().copied().flat_map(|face| match face {
    |                                    ^^^^^^^^ borrow the array with `&` or call `.iter()` on it to iterate over it
    |
    = help: the trait `std::iter::Iterator` is not implemented for `[geometry::Edge; 3]`
    = note: arrays are not iterators, but slices like the following are: `&[1, 2, 3]`
    = note: required because of the requirements on the impl of `std::iter::IntoIterator` for `[geometry::Edge; 3]`

error[E0277]: `[geometry::Edge; 3]` is not an iterator
   --> src\geometry.rs:176:35
    |
176 |     pub fn edges<'a>(&'a self) -> impl Iterator<Item = Edge> + 'a {
    |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ borrow the array with `&` or call `.iter()` on it to iterate over it
    |
    = help: the trait `std::iter::Iterator` is not implemented for `[geometry::Edge; 3]`
    = note: arrays are not iterators, but slices like the following are: `&[1, 2, 3]`
    = note: required because of the requirements on the impl of `std::iter::IntoIterator` for `[geometry::Edge; 3]`
    = note: the return type of a function must have a statically known size
```